### PR TITLE
adding tests for users & posts

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: instagram_own_version/settings.py [hunk 1]
```diff
@@ -49,7 +49,9 @@
 REST_FRAMEWORK = {
     "DEFAULT_AUTHENTICATION_CLASSES": [
         "rest_framework.authentication.TokenAuthentication",
-    ]
+    ],
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.LimitOffsetPagination",
```
Comment (revimi) [inline instagram_own_version/settings.py]: Pagination is not related to this PR. Please create a separate one

File: posts/tests.py [hunk 2]
```diff
@@ -1,3 +1,69 @@
...
+        return {"pk": User.objects.get(username=username).id}
+
+    def authenticate(self, login_data):
+        response = self.client.post(reverse("token"), login_data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        user = User.objects.last()
+        token = Token.objects.get(user=user)
+        self.assertEqual(response.data["token"], token.key)
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + token.key)
```
Comment (revimi) [inline posts/tests.py]: We can move common methods to a parent class

Comment (revimi) [reply posts/tests.py]: Here and in other tests

**Review by revimi — CHANGES_REQUESTED** [review]

File: posts/tests.py (hunk 2 above)
Comment (marinaRomanchuk) [reply posts/tests.py]: Parent classes were created

File: instagram_own_version/settings.py (hunk 1 above)
Comment (marinaRomanchuk) [reply instagram_own_version/settings.py]: Removed

File: posts/tests.py [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+            "password": "daniel1719",
+        }
+        self.client.post(reverse("signup"), self.signup_data)
+        self.client.post(reverse("signup"), self.signup_data_another)
+        post = Post(
+            photo="data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t"
+            "/f3//Ub//ge8WSLf/rhf/",
+            description="picture",
+            user=User.objects.get(username=self.signup_data.get("username")),
+        )
```
Comment (revimi) [inline posts/tests.py]: Why do you create users with the API call but create a post directly via the database?

Comment (revimi) [reply posts/tests.py]: Also, let's better use IDs to get entities

File: posts/tests.py [hunk 4]
```diff
@@ -1,3 +1,61 @@
...
+            ),
+            following=User.objects.get(username=self.signup_data.get("username")),
+        )
+        follower.save()
+
+    def authenticate(self, login_data):
+        response = self.client.post(reverse("token"), login_data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.user = User.objects.get(username=login_data.get("username"))
```
Comment (revimi) [inline posts/tests.py]: Let's better use IDs to get entities

File: posts/tests.py [hunk 5]
```diff
@@ -1,3 +1,61 @@
...
+            photo="data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t"
+            "/f3//Ub//ge8WSLf/rhf/",
+            description="picture",
+            user=User.objects.get(username=self.signup_data.get("username")),
+        )
+        post.save()
+        self.timestamp = post.timestamp
+
+        follower = Followers(
+            follower=User.objects.get(
```
Comment (revimi) [inline posts/tests.py]: Let's better use IDs to get entities

File: posts/tests.py [hunk 6]
```diff
@@ -1,3 +1,61 @@
...
+    def test_str(self):
+        post = Post.objects.get(timestamp=self.timestamp)
+        self.assertEqual(str(post), f"{post.user.username}'s post")
+
+
+class ListPostTest(PostTest):
+    def test_can_retrieve_self_list(self):
+        self.authenticate(self.signup_data)
+        response = self.client.get(reverse("posts-list"), {"user_id": self.user.id})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (revimi) [inline posts/tests.py:69]: Let's check the content of the response

File: posts/tests.py [hunk 7]
```diff
@@ -1,3 +1,61 @@
...
+
+        self.user = User.objects.get(username=login_data.get("username"))
+        token = Token.objects.get(user=self.user)
+        self.assertEqual(response.data["token"], token.key)
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + token.key)
+
+
+class RetrievePostTest(PostTest):
+    def test_str(self):
+        post = Post.objects.get(timestamp=self.timestamp)
```
Comment (revimi) [inline posts/tests.py]: The test class is called `RetrievePostTest` but you are not retreiving the object via API

File: users/tests.py [hunk 8]
```diff
@@ -1,3 +1,117 @@
...
+
+    def test_can_create_user(self):
+        response = self.client.post(reverse("signup"), self.signup_data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_can_not_create_user(self):
+        response = self.client.post(reverse("signup"), self.signup_data_another)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.test_can_create_user()
```
Comment (revimi) [inline users/tests.py]: Not sure it's a good idea to call one test from another as it may change and broke the current one

File: users/tests.py [hunk 9]
```diff
@@ -1,3 +1,117 @@
...
+        self.authenticate(self.signup_data)
+        response = self.client.get(
+            reverse(
+                "user",
+                kwargs=self.get_argument(self.signup_data_another.get("username")),
+            )
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.data.pop("date_of_birth")
```
Comment (revimi) [inline users/tests.py]: Let's probably have another data dict for another user

**Review by revimi — CHANGES_REQUESTED** [review]

File: posts/tests.py (hunk 3 above)
Comment (marinaRomanchuk) [reply posts/tests.py]: Done, now posts also can be created with the API

File: posts/tests.py (hunk 4 above)
Comment (marinaRomanchuk) [reply posts/tests.py]: Done

File: posts/tests.py (hunk 5 above)
Comment (marinaRomanchuk) [reply posts/tests.py]: Done

File: posts/tests.py (hunk 7 above)
Comment (marinaRomanchuk) [reply posts/tests.py]: Renamed

File: users/tests.py (hunk 9 above)
Comment (marinaRomanchuk) [reply users/tests.py]: Done

File: users/tests.py (hunk 8 above)
Comment (marinaRomanchuk) [reply users/tests.py]: Changed

File: posts/tests.py (hunk 6 above)
Comment (marinaRomanchuk) [reply posts/tests.py:69]: Do we need to check photo(decode base64 into url)?

Comment (revimi) [reply posts/tests.py:69]: Yes let's try

File: users/tests.py [hunk 10]
```diff
@@ -1,3 +1,117 @@
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        user = User.objects.get(username=login_data.get("username"))
+        token = Token.objects.get(user=user)
+        self.assertEqual(response.data["token"], token.key)
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + token.key)
+
+
+class CreateUserTest(APITestCase):
+    def setUp(self):
```
Comment (revimi) [inline users/tests.py:52]: Let's better use snake case
```suggestion
    def setup(self):
```

**Review by revimi — APPROVED** [review]
Approved with few comments

**Pull request merged**